### PR TITLE
bug 1691999: update socorro-submitter dev environment

### PR DESCRIPTION
**Pull request by willkg**
This fixes a lot of issues with the dev environment:

1. updates localstack docker image and fixes configuration for it
2. updates socorro_collector image and fixes configuration and test
   outcomes due to recent changes
3. updates lambci/lambda image and requirements-runtime.txt based on the
   new image
4. updates python:3.8 docker image and fixes dependency conflicts
   because we're using the new pip and dependency resolver
5. updates `requirements-dev.txt` adding `awscli` and friends
6. fixes `userid` and `groupid` for app user which fixes permissions issues
   on Linux
7. adds `set -euo pipefail` to top of all bash scripts and fixes
   unassigned variables, exit code handling, and other things
8. improves `bin/integration_test.sh` making it easier to read/maintain

### CODE REVIEW

File: bin/aws_s3.sh [hunk 1]
```diff
@@ -24,8 +26,8 @@ if [[ ! -f /tmp/.aws/credentials ]]
 then
     cat > /tmp/.aws/credentials <<EOF
 [default]
-aws_access_key_id = ${AWS_ACCESS_KEY_ID}
-aws_secret_access_key = ${AWS_SECRET_ACCESS_KEY}
+aws_access_key_id = ${SUBMITTER_S3_ACCESS_KEY}
+aws_secret_access_key = ${SUBMITTER_S3_SECRET_ACCESS_KEY}
```
Comment (willkg) [inline bin/aws_s3.sh:30]: Turns out we were using the wrong variables. Oops.

File: bin/integration_test.sh [hunk 2]
```diff
@@ -8,7 +8,15 @@
...
-set -e
+set -euo pipefail
+
+newtest() {
+    echo ""
+    echo "========================================================================"
+    echo ">>> $1"
+    echo "========================================================================"
+    echo ""
+}
```
Comment (willkg) [inline bin/integration_test.sh:19]: This makes it easier to see individual tests and follow the ridiculous integration test output.

File: bin/integration_test.sh [hunk 3]
```diff
@@ -22,7 +30,7 @@ reportfail() {
     echo ""
 }
 
-trap reportfail 0
+trap "reportfail ." 0
```
Comment (willkg) [inline bin/integration_test.sh:33]: Funny story--because `reportfail` looks at `$1`, if the script failed, this would cause it to fail on the trap because it wasn't passing in an argument.

File: bin/integration_test.sh [hunk 4]
```diff
@@ -56,6 +64,10 @@ docker-compose up -d antenna
 # from inside the containers.
 ./bin/wait.sh localhost 8888
 
+# Remove anything in the bucket that Antenna generated at startup so we have an
+# empty bucket to work with.
+docker-compose run -u "${HOSTUSER}" test ./bin/aws_s3.sh rm "${DESTBUCKET}" --recursive
```
Comment (willkg) [inline bin/integration_test.sh:69]: This fixes an edge cases where running `bin/integration_test.sh` right after building things but before starting any of the services caused the test to fail.

File: bin/integration_test.sh [hunk 5]
```diff
@@ -79,17 +92,21 @@ then
     exit 1
 fi
 
-# Make sure nothing is in the dest bucket and thus nothing got submitted
+# Make sure nothing is in the dest bucket (except the test/ thing that antenna
+# posts at startup to test it can save to the bucket) and thus nothing got
+# submitted
```
Comment (willkg) [inline bin/integration_test.sh:97]: I can revert this change.

File: bin/setup_integration_test.sh [hunk 6]
```diff
@@ -6,18 +6,10 @@
...
-if [ -z "${SUBMITTER_S3_BUCKET}" ]; then
-    echo "SUBMITTER_S3_BUCKET is not set."
-    exit 1;
-fi
+set -euo pipefail
 
-if [ -z "${CRASHSTORAGE_BUCKET_NAME}" ]; then
-    echo "CRASHSTORAGE_BUCKET_NAME is not set."
-    exit 1;
-fi
```
Comment (willkg) [inline bin/setup_integration_test.sh:17]: These just checked to make sure the variables were defined. But we're doing that with the `set -euo pipefail` so we can nix these.

File: bin/setup_integration_test.sh [hunk 7]
```diff
@@ -6,18 +6,10 @@
...
-
-./bin/aws_s3.sh rb --force s3://${SUBMITTER_S3_BUCKET}/ 2>&1 > /dev/null
+./bin/aws_s3.sh rb --force s3://${SUBMITTER_S3_BUCKET}/ || true
 ./bin/aws_s3.sh mb s3://${SUBMITTER_S3_BUCKET}/
 
-./bin/aws_s3.sh rb --force s3://${CRASHSTORAGE_BUCKET_NAME}/ 2>&1 > /dev/null
+./bin/aws_s3.sh rb --force s3://${CRASHSTORAGE_BUCKET_NAME}/ || true
```
Comment (willkg) [inline bin/setup_integration_test.sh:14]: This was printing stuff out anyhow, so I nixed all the out/err redirection which apparently wasn't working. Further, because this script now dies if there's a non-zero exit code, we add `|| true`.

File: bin/wait.sh [hunk 8]
```diff
@@ -23,8 +25,9 @@ do
         exit 1
     fi
 
-    nc -z ${HOST} ${PORT}
-    if [ $? -eq 0 ]
+    ERR=0
+    nc -v -z ${HOST} ${PORT} || ERR=1
+    if [ "${ERR}" == "0" ]
```
Comment (willkg) [inline bin/wait.sh:30]: This was tricky. We want `nc` to die and then retry this script, so we need to figure out when it died without having the non-zero exit code kill the script entirely. This was the best way I could figure out to do it.

File: docker-compose.yml [hunk 9]
```diff
@@ -33,27 +34,27 @@ services:
 
   # https://hub.docker.com/r/localstack/localstack/
   # localstack running a fake S3
-  localstack-s3:
-    image: localstack/localstack:0.10.8
+  localstack:
+    image: localstack/localstack:0.12.2
```
Comment (willkg) [inline docker-compose.yml:38]: I updated localstack and the new version does all the services in a single container with a single port, so I made a bunch of naming and configuration fixes.

File: docker-compose.yml [hunk 10]
```diff
@@ -33,27 +34,27 @@ services:
...
     env_file: docker/antenna.env
     depends_on:
-      - localstack-s3
+      - localstack
     expose:
       - 8000
     ports:
       - "8888:8000"
-    command: ./bin/run_web.sh
+    command: web
```
Comment (willkg) [inline docker-compose.yml:60]: I'm not sure if the older version didn't set an entrypoint or what, but this runs the web command in the entrypoint in the service now.

File: fakedata_source/v2/raw_crash/111/20180313/11107bd0-2d1c-4865-af09-80bc00180313 [hunk 11]
```diff
@@ -1 +1 @@
-{"MinidumpSha256Hash": "ae0666f161fed1a5dde998bbd0e140550d2da0db27db1d0e31e370f2bd366a57", "ProductName": "Firefox", "ReleaseChannel": "beta", "Version": "60.0b2", "dump_checksums": {"upload_file_minidump": "ae0666f161fed1a5dde998bbd0e140550d2da0db27db1d0e31e370f2bd366a57"}, "legacy_processing": 0, "submitted_timestamp": "2018-04-11T16:25:35.023214+00:00", "throttle_rate": 100, "timestamp": 1523463935.0180907, "type_tag": "bp", "uuid": "11107bd0-2d1c-4865-af09-80bc00180313"}
+{"MinidumpSha256Hash": "ae0666f161fed1a5dde998bbd0e140550d2da0db27db1d0e31e370f2bd366a57", "ProductName": "Firefox", "ReleaseChannel": "beta", "Version": "60.0b2", "collector_notes": [], "dump_checksums": {"upload_file_minidump": "ae0666f161fed1a5dde998bbd0e140550d2da0db27db1d0e31e370f2bd366a57"}, "payload": "multipart", "payload_compressed": "0", "submitted_timestamp": "2018-04-11T16:25:35.023214+00:00", "uuid": "11107bd0-2d1c-4865-af09-80bc00180313"}
```
Comment (willkg) [inline fakedata_source/v2/raw_crash/111/20180313/11107bd0-2d1c-4865-af09-80bc00180313:1]: This reflects a bunch of changes to Antenna over the last 6 months. I added a few new fields and removed a bunch of legacy fields that weren't interesting.
